### DISCUSSION TITLE
rebase-todo files

### DISCUSSION THREAD

**Discussion by nikita-skobov** [libgit2/libgit2#5616] [General] [answered]
In git when you do an interactive rebase and stop it midway with a break or edit, you can go back to the rebase todo file via: `.git/rebase-merge/git-rebase-todo`

I belived this file also appears when you do a regular rebase that results in a conflict.

I noticed when running a basic example of doing a rebase that it doesn't seem to use or make such a file. Is there a way to get libgit2 to make and use such a file, or otherwise how could I use libgit2 to try to do some basic interactive rebases?

**Comment by ethomson** [comment, marked as answer]
There are three types of rebase: `rebase-am` (using patch files), `rebase-merge` (using cherry-pick) and `rebase-interactive` (using the list of instructions).  At the moment, we support only `rebase-merge`.  Our rebase machinery was built with the abstraction points to support interactive, but this has not been implemented.  At the moment libgit2 will not start an interactive rebase, nor support any interactive rebase in-progress.

**Reply by nikita-skobov** [reply]
ok thanks for the quick response.

If I wanted some basic `rebase-interactive` functionality (I just need to either pick, or skip commits), is it worth considering implementing that via a series of cherry picks in `libgit2`, or am I better off using the CLI for such a task?
